### PR TITLE
Migrate access/msteams to slog + add debug logs

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR does two things:
- migrate the msteams access plugin to slog
- add a lot of debug log statements to make troubleshooting easier

changelog: Improve the msteams access plugin debug logging.

### CODE REVIEW

File: integrations/lib/logger/logger.go [hunk 1]
```diff
@@ -90,6 +98,86 @@ func Setup(conf Config) error {
...
+			ConfiguredFields: configuredFields,
+		}))
+		slog.SetDefault(slogLogger)
+	default:
+		return nil, trace.BadParameter("unsupported log output format : %q", conf.Format)
+	}
+
+	return slogLogger, nil
+}
+
```
Comment (hugoShaka) [inline integrations/lib/logger/logger.go:180]: This logic is a port from the `config.applyLogConfig` logic. The original `applyLogConfig` function depends on `servicecfg.Config` and I did not want to refactor this part as well (the plugins don't have a `servicecfg.Config`.

**Review by r0mant — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

File: integrations/access/msteams/app.go [hunk 2]
```diff
@@ -67,8 +78,7 @@ func NewApp(conf Config) (*App, error) {
 
 // Run starts the main job process
 func (a *App) Run(ctx context.Context) error {
-	log := logger.Get(ctx)
-	log.Info("Starting Teleport MS Teams Plugin")
+	a.log.InfoContext(ctx, "Starting Teleport MS Teams Plugin", "plugin_name", pluginName)
```
Comment (hugoShaka) [inline integrations/access/msteams/app.go:81]:
```suggestion
	a.log.InfoContext(ctx, "Starting Teleport MS Teams Plugin")
```

Plugin name is always set in `NewApp()`

File: integrations/access/msteams/app.go [hunk 3]
```diff
@@ -294,15 +298,16 @@ func (a *App) onWatcherEvent(ctx context.Context, event types.Event) error {
...
-	log := logger.Get(ctx)
-
 	id := req.GetName()
 	data := pd.AccessRequestData{
 		User:          req.GetUser(),
 		Roles:         req.GetRoles(),
 		RequestReason: req.GetRequestReason(),
 	}
 
+	log := a.log.With("req_id", id)
```
Comment (hugoShaka) [inline integrations/access/msteams/app.go:308]:
```suggestion
	log := a.log.With("request_id", id)
```

Consistency

File: integrations/access/msteams/app.go [hunk 4]
```diff
@@ -242,48 +247,47 @@ func (a *App) onWatcherEvent(ctx context.Context, event types.Event) error {
 
 	op := event.Type
 	reqID := event.Resource.GetName()
-	ctx, _ = logger.WithField(ctx, "request_id", reqID)
+	log := a.log.With("req_id", reqID)
```
Comment (hugoShaka) [inline integrations/access/msteams/app.go:250]:
```suggestion
	log := a.log.With("request_id", reqID)
```

**Review by espadolini — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]
Thanks Hugo!

**Pull request merged**